zipline/assets/roll_finder.py

```diff
@@ -191,10 +191,15 @@ def _active_contract(self, oc, front, back, dt):
         # instead of to 'dt' is because we need to get each contract's volume
         # on the previous day, so we need to make sure the back contract exists
         # on 'prev' in order to call 'get_value' below.
-        if dt > front_contract.auto_close_date:
+        if dt > min(front_contract.auto_close_date, front_contract.end_date):
             return back
         elif prev < back_contract.start_date:
             return front
+        elif prev > back_contract.end_date:
+            # This is an extreme edge case, but it is possible for the back
+            # contract to end before the front contract. At this point we know
+            # the front contract still exists, so just return that.
+            return front
```

Comment:
Yes, not sure how I didn't think of that after I literally just looked at the `can_trade` code 😆